lab7/app/src/main/java/com/example/project2/MainActivity.java

```diff
@@ -0,0 +1,146 @@
+package com.example.project2;
+
+import androidx.annotation.RequiresApi;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.app.KeyguardManager;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.os.Bundle;
+import android.preference.Preference;
+import android.security.keystore.KeyGenParameterSpec;
+import android.security.keystore.KeyProperties;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.io.IOException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.util.Base64;
+import java.util.Enumeration;
+
+import javax.crypto.Cipher;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+    public static final String SHARED_PREFS = "sharedPrefs";
+    public Button savebtn,showpinbtn;
+    public KeyPair pair;
+    public EditText textbox;
+    public TextView status;
+    public  KeyStore ks;
+    @Override
+    protected void onCreate(Bundle savedInstanceState)  {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        KeyguardManager km = (KeyguardManager)getApplicationContext().getSystemService(Context.KEYGUARD_SERVICE);
+        assert km != null;
+        if(km.isKeyguardSecure())
+            Toast.makeText(getApplicationContext(), "Phone is Secure!", Toast.LENGTH_LONG).show();
+        else {
+            Toast.makeText(getApplicationContext(), "Phone is Unlocked!", Toast.LENGTH_LONG).show();
+            finish();
+        }
+        try {
+            ks = KeyStore.getInstance("AndroidKeyStore");
+            ks.load(null);
+            Enumeration<String> aliases = ks.aliases();
+            pair = generateKeyPair();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
printStackTrace is not used in the Android world, we prefer Log with warning or error level. Please replace it everywhere you use it.